lib/strdup.c

```diff
@@ -50,6 +55,35 @@ char *curlx_strdup(const char *str)
 }
 #endif
 
+#ifdef WIN32
+/***************************************************************************
+ *
+ * Curl_wcsdup(source)
+ *
+ * Copies the 'source' wchar string to a newly allocated buffer (that is
+ * returned).
+ *
+ * Returns the new pointer or NULL on failure.
+ *
+ ***************************************************************************/
+wchar_t *Curl_wcsdup(const wchar_t *src)
+{
+  size_t length = wcslen(src);
+  size_t lengthnul;
+  size_t lengthbytes;
+
+  if (SIZE_MAX - length < 1)
```

Comment:
These overflow checks are probably unnecessary, because presumably the wchar string has a size sitting in size_t.  I don't know what your standards are for checks like these.